src/w3d/renderer/textureloader.cpp

```diff
@@ -81,9 +81,18 @@ w3dtexture_t Load_Compressed_Texture(
 void LoaderThreadClass::Thread_Function()
 {
     while (m_isRunning) {
-        if (!g_backgroundQueue.Empty()) {
+        bool backgroundQueueEmpty;
+        {
             FastCriticalSectionClass::LockClass background_lock(g_backgroundCritSec);
```

Comment:
Originally takes both locks at the same time. This can be risky and introduce ABA deadlock problem. This can be avoided by limiting scope.

Could perhaps deadlock with `TextureLoader::Request_Foreground_Loading`.